### PR TITLE
Lambda lookup

### PR DESCRIPTION
**Pull request by maartenvanderhoef**
## What
Adding a Lambda which does a lookup of the ECS Container information. container_image renamed to bootstrap_container_image to make it more clear that it's just to ignite. Added the  force_bootstrap_container_image to force the ECS Service with the given image no matter what.

## Why
Regular tf datasources have as downside that they won't work at bootstrap, this lambda does!

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -215,8 +215,21 @@ variable "default_capacity_properties_deployment_minimum_healthy_percent" {
...
+# live_task_lookup_type
+# This module is capable of working around the state drift when external CICD deploys to ECS
+# By default a Lambda takes care of looking up the current container information, when the type is set to `lambda`
+# When the type is set to `datasource` regular terraform datasources are used to look-up the current container
+# Downside of datasource is that it cannot be used for bootstrapping
+variable "live_task_lookup_type" {
+  default = "lambda"
+}
+
 # image_url defines the docker image location
```
[the remark this answers is not in the document]
Comment (maartenvanderhoef) [reply variables.tf]: @jonboulle wdym?

File: modules/live_task_lookup/output.tf [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  empty_lookup = {
+    environment        = ""
+    image              = ""
+    cpu                = ""
+    memory             = ""
+    memory_reservation = ""
+    task_revision      = ""
+  }
+
+  # Lambda invoke returns a map, but as it's inside a conditional coalescelist make sure it can be looked up when it does not exist
```
[the remark this answers is not in the document]
Comment (maartenvanderhoef) [reply modules/live_task_lookup/output.tf]: 👯‍♂️

File: modules/live_task_lookup/variables.tf [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+variable "ecs_cluster_id" {}
+
+# lambda_lookup_role_arn sets the role arn of the lookup_lambda
+variable "lambda_lookup_role_arn" {}
+
+# lookup_type sets the type of lookup, either 
+# * lambda - works during bootstrap and after bootstrap
+# * datasource - uses terraform datasources ( aws_ecs_service ) which won't work during bootstrap
+variable "lookup_type" {
+  default = "lambda"
```
[the remark this answers is not in the document]
Comment (maartenvanderhoef) [reply modules/live_task_lookup/variables.tf]: Good 1.

File: modules/live_task_lookup/main.tf [hunk 4]
```diff
@@ -1,49 +1,48 @@
...
+}
+JSON
+}
+
+#
+# lookup_type datasource
+#
 data "aws_ecs_service" "lookup" {
-  count        = "${var.create ? 1 : 0 }"
+  count        = "${var.create && var.lookup_type == "datasource" ? 1 : 0 }"
```
[the remark this answers is not in the document]
Comment (maartenvanderhoef) [reply modules/live_task_lookup/main.tf]: I think I leave it for now, if it get's bigger than we'll have to do that.

File: variables.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonboulle) [reply variables.tf]: it's no longer called image_url

**Pull request merged**